examples/sunpy_other_packages/reprojection_heliographic_stonyhurst.py

```diff
@@ -0,0 +1,69 @@
+"""
+===========================
+Creating a Heliographic Map
+===========================
+
+In this example we use the `reproject` package to make a heliographic image
+from an AIA image.
+
+You will need to install reproject to run this example.
+"""
+# sphinx_gallery_thumbnail_number = 2
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+from astropy.wcs import WCS
+
+from reproject import reproject_interp
+
+import sunpy.data.sample
+import sunpy.map
+
+###############################################################################
+# We start with the sample data
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_193_IMAGE)
+
+fig = plt.figure()
+ax = plt.subplot(projection=aia_map)
+aia_map.plot(ax)
```

Comment:
I have a suspicion that we have a policy of not using `peek` in examples and stuff. I can't remember exactly where we ended up on this. Maybe @ayshih has thoughts.

Reply:
Heh, of course I have thoughts!  Yes, we essentially have an unwritten policy of not using `peek()` in examples when the objective is to create a plot, because that's what `plot()` is for.  The exact nature of `peek()` – its use cases, what it produces, and even its name – is still subject to discussion and change, and so `peek()` should not be relied on to create plots.

Reply:
But isn't the whole point of `peek()` to quickly have a glance at a plot, which is the use case here? If an un-written policy is not to use peek, then it should be a written policy in the `peek()` docs...

Reply:
Not exactly.  You may recall past discussions on `peek()` (e.g., https://github.com/sunpy/sunpy/pull/3103#issuecomment-494569464).  Here's what the docstring for Map's `peek()` says:

> Displays a graphical overview of the data in this object for user evaluation. For the creation of plots, users should instead use the `plot` method and Matplotlib’s pyplot framework.

While this "graphical overview" does (currently) take the form of a Matplotlib plot, we don't want to inadvertently teach people to use `peek()` instead of using `plot()` when creating plots, so that's why we want to refrain from using `peek()` in examples.

`peek()` is great – I use it all the time – but it needs to evolve into its own thing beyond just a shorthand for `plot()`.

Reply:
I'm still a bit confused - what's the difference between a 'graphical overview' and a 'plot'?

Reply:
I'm trying to update the [example gallery section of the developer's guide](https://docs.sunpy.org/en/latest/dev_guide/example_gallery.html#) and [example template](https://github.com/sunpy/sunpy/blob/master/examples/example_template/example_template.py). Are there any other unofficial policies, besides not using `peek()`, that I should include?

Reply:
> I'm still a bit confused - what's the difference between a 'graphical overview' and a 'plot'?

Think of the difference between the text `__repr__()` of a large `ndarray` and converting the `ndarray` contents to a string.  I argue that `peek()` should be a graphical analogue to `__repr__()`.  In my mind, it'd be fine if the "plot" aspect of `peek()` were to be diminished or abbreviated.  There's more to a `Map` object than how it plots; for example, the graphical overview can/should include important metadata or data characteristics.  If you want to see how the object plots, use `plot()`.

> Are there any other unofficial policies, besides not using `peek()`, that I should include?

Not that I'm aware of.  If/when `peek()` evolves into something more than it is, I think it'd be totally fine to include in examples (in the same manner that we show text `__repr__()` output).  It's just that while it currently produces output that is just like `plot()` output, people will (incorrectly) learn it as a shorthand to produce plots.

Reply:
> 
> Think of the difference between the text `__repr__()` of a large `ndarray` and converting the `ndarray` contents to a string. I argue that `peek()` should be a graphical analogue to `__repr__()`. In my mind, it'd be fine if the "plot" aspect of `peek()` were to be diminished or abbreviated. There's more to a `Map` object than how it plots; for example, the graphical overview can/should include important metadata or data characteristics. If you want to see how the object plots, use `plot()`.

Okay, but showing the image as part of `peek()` is presumably never going to go away? The purpose of this bit of code in the example is to take a quick look at what's in the `Map` object, not to create a scientific plot, so I'd argue that `peek()` is definitely appropriate in this place in the example.

Reply:
I agree that if `peek()` were already doing something tangibly different from `plot()`, it'd be the appropriate method to call here.  Given its current output, though, we know that users will get the wrong idea and learn it as a shorthand for `plot()`, and then that will have to be un-learned later.